diff --git a/android/build.gradle b/android/build.gradle
index 8e882e3dfadd3724ff8bd9e0bf6a2dc9a0910ad1..a93ebbbd369d48d8bc8483ff4f892ceddaacfda5 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -22,6 +22,6 @@ android {
 
 dependencies {
   coreLibraryDesugaring 'com.android.tools:desugar_jdk_libs:2.1.5'
-  implementation 'org.eclipse.jgit:org.eclipse.jgit:6.2.0.202206071550-r'
+  implementation 'org.eclipse.jgit:org.eclipse.jgit:6.10.0.202406032230-r'
   implementation "com.squareup.okhttp3:okhttp:4.12.0"
 }
diff --git a/android/src/main/java/expo/modules/externalstorage/GitNetwork.kt b/android/src/main/java/expo/modules/externalstorage/GitNetwork.kt
index c70f061d593109fbb89860c0cca6b8ba896fe551..480a17b1be4516eef5280e1b52b2439dd946f488 100644
--- a/android/src/main/java/expo/modules/externalstorage/GitNetwork.kt
+++ b/android/src/main/java/expo/modules/externalstorage/GitNetwork.kt
@@ -133,6 +133,10 @@ internal object GitNetwork {
         cloneCommand.setBranchesToClone(listOf("refs/heads/$branch"))
       }
 
+      if (depth > 0) {
+        cloneCommand.setDepth(depth)
+      }
+
       if (noTags) {
         cloneCommand.setNoTags()
       }
